R/irods-conf.R

```diff
@@ -17,20 +23,19 @@ create_config_dir <- function(path_to_config_dir) {
 path_to_irods_conf <- function() {
   path_to_config_dir <- rappdirs::user_config_dir("rirods")
   create_config_dir(path_to_config_dir)
```

Comment:
In `dir.create(path_to_config_dir)` (in the definition of `create_config_dir()`) better to add `recursive=TRUE`. When testing this in our test environments some directory at the top didn't exist and therefore this function failed.

Reply:
Then the problem seems to be here: `rappdirs::user_config_dir("rirods")`, as this should ensure to find the config directory on most OS. But it can be mitigated with `recursive`.

Reply:
I add the test below. Does this replicate the problem?

```
test_that("create iRODS configure directory works", {
  # store conf file
  tmp <- tempfile()
  file.copy(path_to_irods_conf(), tmp)
  unlink(path_to_irods_conf())
  pt <- rappdirs::user_config_dir("rirods")
  # add accidental none existing directory
  pt <- file.path(dirname(pt), "notexist", "rirods")
  create_config_dir(pt)
  unlink(pt)
  # restore conf file
  file.copy(tmp, path_to_irods_conf())
})
```

Reply:
Yes, although I find it weird to write a test to test `create_config_dir()` in which you call a function that depends on `create_config_dir()` (nl. `path_to_irods_conf()`). Maybe something like this:

```r
test_that("create iRODS configuration directory works", {
  tmp_dir <- tempdir()
  expect_true(dir.exists(tmp_dir))
  unlink(tmp_dir, force=TRUE, recursive=TRUE)
  expect_false(dir.exists(tmp_dir)) # just making sure

  # test create_config_dir()
  dir_to_create <- file.path(tmp_dir, "rirods")
  create_config_dir(dir_to_create)
  expect_true(dir.exists(dir_to_create))

  # test path_to_irods_conf()
  expect_true(file.exists(path_to_irods_conf()))
})
```

Reply:
The true problem here is `rappdirs::user_config_dir("rirods")`, where it seems you stumbled on an edge case not covered by this function. I think the following is claimed to be true for the OS shown below. Bu apparently this is not the case on your system.

```
# On unix
dir.exists(rappdirs::user_config_dir("something")) # FALSE
dir.exists(dirname(rappdirs::user_config_dir("something"))) # TRUE
# On windows
dir.exists(rappdirs::user_config_dir("something")) # FALSE
dir.exists(dirname(rappdirs::user_config_dir("something"))) # FALSE
dir.exists(dirname(dirname(rappdirs::user_config_dir("something")))) # TRUE
``` 
So I guess the best course of action is maybe to remove the whole function and just use `dir.create('path_to_conf', recursive=TRUE)'. This way the directory is always created independent of how many parents `rappdirs` claims there to be.

Reply:
Yes, that makes sense.

Reply:
I still reconsidered. I think this function is still nice to keep:

```
create_config_dir <- function(path_to_config_dir) {
  if (!dir.exists(path_to_config_dir)) {
    dir.create(path_to_config_dir, recursive = TRUE)
  }
}

```

The following test would test it (although maybe a bit overkill) without destroying the test-configuration file:

```
test_that("create iRODS configuration directory works", {
  tmp_dir <- rappdirs::user_config_dir("mock_config_dir")
  if (!dir.exists(tmp_dir))
    dir.create(tmp_dir, recursive = TRUE)
  expect_true(dir.exists(tmp_dir))
  unlink(tmp_dir, force=TRUE, recursive=TRUE)
  expect_false(dir.exists(tmp_dir)) # just making sure

  # test create_config_dir()
  expect_invisible(create_config_dir(tmp_dir))
  expect_true(dir.exists(tmp_dir))
})

```